ms2deepscore/BinnedSpectrum.py

```diff
@@ -0,0 +1,39 @@
+class BinnedSpectrum:
+    """Binned spectrum for use with MS2DeepScore."""
+    def __init__(self, binned_peaks: dict, metadata: dict):
+        self.peaks = binned_peaks
+        self._metadata = metadata
```

Comment:
As I mentioned before, I would prefer having attributes of the `BinnedSpectrum` class over the `metadata` dict. It is unclear what goes into metadata, what should go in there and what can I expect to be in there. It creates unnecessary nestedness. It makes the code more complex because you need the getters and setters. You get some flexibility back for it, but I am not sure if you really need that.

Reply:
You are right that attributes would be cleaner here. In particular now that we hardly need a lot of metadata entries yet. But I have chosen this for a start because it is analogue to `Spec2Vec` and `matchms` which might have some advantages when using those things together. I will hence leave it for now, but let's check on this later.